modular_skyrat/modules/fleshmind/code/make_ert.dm

```diff
@@ -0,0 +1,121 @@
+	if(!ert_type)
+		return
+
+	var/datum/ert/created_ert_datum = new ert_type
+
+	created_ert_datum.teamsize = teamsize
+	created_ert_datum.mission = mission_objective_override
+	created_ert_datum.polldesc = poll_description
+	created_ert_datum.enforce_human = enforce_human
+	created_ert_datum.opendoors = open_armory_doors
+	created_ert_datum.leader_experience = leader_experience
+	created_ert_datum.random_names = random_names
+	created_ert_datum.notify_players = notify_players
+	created_ert_datum.code = code
+
+	var/list/spawnpoints
+
+	if(spawnpoint_override)
+		spawnpoints = spawnpoint_override
+	else
+		spawnpoints = GLOB.emergencyresponseteamspawn
+
+	if(!LAZYLEN(spawnpoints))
+		CRASH("make_ert had no ERT spawnpoints to choose from!")
+
+	var/list/mob/dead/observer/candidates = poll_ghost_candidates("Do you wish to be considered for [created_ert_datum.polldesc]?", "deathsquad")
+
+	if(!LAZYLEN(candidates))
+		return FALSE
+
+	//Create team
+	var/datum/team/ert/ert_team = new created_ert_datum.team
+	if(created_ert_datum.rename_team)
+		ert_team.name = created_ert_datum.rename_team
+
+	//Assign team objective
+	var/datum/objective/mission_objective = new ()
+	mission_objective.team = ert_team
+	mission_objective.explanation_text = created_ert_datum.mission
+	mission_objective.completed = TRUE
+	ert_team.objectives += mission_objective
+	ert_team.mission = mission_objective
+
+	var/mob/dead/observer/earmarked_leader
+	var/leader_spawned = FALSE // just in case the earmarked leader disconnects or becomes unavailable, we can try giving leader to the last guy to get chosen
+
+	if(created_ert_datum.leader_experience)
+		var/list/candidate_living_exps = list()
+		for(var/mob/dead/observer/potential_leader as anything in candidates)
+			candidate_living_exps[potential_leader] = potential_leader.client?.get_exp_living(TRUE)
+
+		candidate_living_exps = sort_list(candidate_living_exps, cmp = /proc/cmp_numeric_dsc)
+		if(LAZYLEN(candidate_living_exps) > ERT_EXPERIENCED_LEADER_CHOOSE_TOP)
+			candidate_living_exps = candidate_living_exps.Cut(ERT_EXPERIENCED_LEADER_CHOOSE_TOP + 1) // pick from the top ERT_EXPERIENCED_LEADER_CHOOSE_TOP contenders in playtime
+		earmarked_leader = pick(candidate_living_exps)
+	else
+		earmarked_leader = pick(candidates)
+
+	var/index = 0
+	var/number_of_agents = min(created_ert_datum.teamsize, LAZYLEN(candidates))
+	var/team_spawned = FALSE
+
+	while(number_of_agents && LAZYLEN(candidates))
+		var/spawnloc = spawnpoints[index + 1]
+		//loop through spawnpoints one at a time
+		index = (index + 1) % spawnpoints.len
+		var/mob/dead/observer/chosen_candidate = earmarked_leader || pick(candidates) // this way we make sure that our leader gets chosen
+		candidates -= chosen_candidate
```

Comment:
use LAZYREMOVE (same goes for other list ops of `candidates` and `candidate_living_exps`